### PR TITLE
simple_msgs::Image cleanup

### LINKED ISSUE #28: Image message fields extension/cleanup
I have several issues with the image data field:

To completely specify positioning of an image in 3D space, it should have an additional vector describing the pixel scaling of each axis - that way, together with the rest of the pose fields, it is possible to fully map from a pixel to 3D space even if there is non-uniform scaling, which is common for some modalities. This might already be stored with the image resolution field, but I cannot really find out what the supposed semantics are due to missing docs. If so, maybe this could be renamed to pixel scaling etc. for clarity, image resolution usually refers to the number of pixels in each dimension...

Furthermore, there are some fields which are unnecessarily signed:
width/height/depth/channels/image_size have no (obvious) need for a sign bit, so I would suggest making them unsigned unless there is some specific meaning related to negative numbers.

### PR DESCRIPTION
**Pull request by SalvoVirga**
Fixes #28 

Fields related to the distance between pixels in the image/volume (aka resolution or spacing) are not called spacing_x_, spacing_y_ and spacing_z_. Relative methods also changed their names.

Image fields should now have the correct size:
width, height and depth are now uint32
data_size is a uint64
num_channels is an ushort

### CODE REVIEW

**Comment by SalvoVirga** [comment]
@TheHugeManatee I invited you to the github group, so that you can also provide a review of this PR

File: examples/src/example_image_subscriber.cpp [hunk 1]
```diff
@@ -33,8 +33,12 @@ void example_callback(const simple_msgs::Image<uint8_t>& i) {
 
   // Get the image dimensions, e.g. 512x512x1.
   const auto dimensions = i.getImageDimensions();
+
   // Build an OpenCV Mat from those.
-  cv::Mat received_img{dimensions[1], dimensions[0], CV_8UC(i.getNumChannels()), img};
+  // We need to cast the image dimensions, since OpenCV takes signed ints :(
+  cv::Mat received_img{static_cast<int>(dimensions[1]), static_cast<int>(dimensions[0]), CV_8UC(i.getNumChannels()),
+                       img};
+  cv::Mat();
```
Comment (TheHugeManatee) [inline examples/src/example_image_subscriber.cpp]: spurious temporary cv::Mat here?

File: msgs/include/simple_msgs/image.hpp [hunk 2]
```diff
@@ -152,30 +152,30 @@ class Image : public GenericMessage {
...
   const T* getImageData() const { return data_.getData(); }
-  int getImageSize() const { return data_size_; }
+  uint64_t getImageSize() const { return data_size_; }
   const Header& getHeader() const { return header_; }
   Header& getHeader() { return header_; }
   const Pose& getImageOrigin() const { return origin_; }
   Pose& getImageOrigin() { return origin_; }
   std::string getImageEncoding() const { return encoding_; }
-  int getNumChannels() const { return num_channels_; }
+  unsigned int getNumChannels() const { return num_channels_; }
```
Comment (TheHugeManatee) [inline msgs/include/simple_msgs/image.hpp]: this should be uint32_t for consistency

**Review by TheHugeManatee — CHANGES_REQUESTED** [review]

File: examples/src/example_image_subscriber.cpp (hunk 1 above)
Comment (SalvoVirga) [reply examples/src/example_image_subscriber.cpp]: ups, a left over from debugging

File: msgs/include/simple_msgs/image.hpp (hunk 2 above)
Comment (SalvoVirga) [reply msgs/include/simple_msgs/image.hpp]: you mean unsigned short?

I changed num_channel to that

 ```unsigned short num_channels_{1};```

I guess that for the number of channels we can use something very tiny.

Comment (SalvoVirga) [reply msgs/include/simple_msgs/image.hpp]: but then why I didn't use uint16_t? :D

**Review by fernandalangsch — APPROVED** [review]

**Review by TheHugeManatee — APPROVED** [review]

**Pull request merged**